app/lib/features/chat/widgets/chats_list.dart

```diff
@@ -122,23 +127,39 @@ class _AnimatedChatsList extends StatefulWidget {
 }
 
 class __AnimatedChatsListState extends State<_AnimatedChatsList> {
-  final GlobalKey<AnimatedListState> _listKey = GlobalKey<AnimatedListState>();
```

Comment:
this is where the actual bug was. keeping the _listKey around but effectively out-of-sync with the actual list meant we were rendering old state and didn't realize it.